### ISSUE TITLE
sharpness detection

### ISSUE THREAD

**Issue by fairynn** [xinario/SAGAN#6]
I wonder why training an independent network for sharpness detection?  Why not calculate the sharpness difference between synthesized images and real images and then add the sharpness loss to objective function?

**Comment by xinario** [comment]
Good question. 

The reason is that the sharpness metric we used from the paper "lbp-based segmentation of defocus blur" is not differentiable. Therefore, in order to use gradient descent methods like Adam to optimize the loss, we resorted to a  dedicated network to do that.
